### PR TITLE
hotdoc: init at 0.12.2

### LINKED ISSUE #98762: hotdoc: needed for updating gst_all_1 derivations
**Project description**
Documentation system, needed now by gst 1.18.0.

**Metadata**

* homepage URL: https://hotdoc.github.io/
* source URL: https://github.com/hotdoc/hotdoc
* license: mit, bsd, gpl2+ , ... lgpl2**Plus**
* platforms: Python?

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

Fix https://github.com/NixOS/nixpkgs/issues/98762 .

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/hotdoc/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  preConfigure =
+    # Cmake is needed to build a c extension CMARK, that's found inside the
+    # source tarball. We add cmake to PATH here, and not in nativeBuildInputs,
+    # since it's not as the main build system of the project.
+  ''
+    export PATH="$PATH:${lib.getBin cmake}/bin"
```
Comment (jtojnar) [inline pkgs/development/tools/hotdoc/default.nix:28]: That does not change the fact that `cmake` is a build dependency. You will just need to use `dontUseCmakeConfigure = true;`.

File: pkgs/development/tools/hotdoc/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  pname = "hotdoc";
+  version = "0.12.2";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "0nypvcqp4b0qfps35f7mz3d90zjh0v5wawpd06ra28q92nxblx17";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
```
Comment (jtojnar) [inline pkgs/development/tools/hotdoc/default.nix:20]: Do not we need `flex`: https://github.com/hotdoc/hotdoc/blob/83d0c70ecb9e0224ec67dcf7c6c5cfee8d049288/setup.py#L389-L422

File: pkgs/development/tools/hotdoc/default.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  ];
+
+  checkInputs = [
+    python3Packages.networkx
+  ];
+
+  meta = {
+    description = "HotDoc: the tastiest API documentation system";
+    homepage = "https://hotdoc.github.io/";
+    license = with lib.licenses; [ lgpl2Plus ];
```
Comment (jtojnar) [inline pkgs/development/tools/hotdoc/default.nix:64]:
```suggestion
    license = with lib.licenses; [ lgpl21Plus ];
```

File: pkgs/development/python-modules/dbus-deviation/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  buildInputs = [
+    setuptools-git
+    setuptools-pep8
+    sphinx
+  ];
+
+  meta = {
+    homepage = "https://pypi.org/project/dbus-deviation/";
+    description = "a project for parsing D-Bus introspection XML and processing it in various ways";
+    license = lib.licenses.lgpl2Plus;
```
Comment (jtojnar) [inline pkgs/development/python-modules/dbus-deviation/default.nix:32]:
```suggestion
    license = lib.licenses.lgpl21Plus;
```

according to https://tecnocode.co.uk/dbus-deviation/

File: pkgs/development/tools/hotdoc/default.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  ];
+
+  buildInputs = [
+    glib
+    json-glib
+    libxml2
+  ];
+
+  checkInputs = [
+    python3Packages.networkx
```
Comment (FRidh) [inline pkgs/development/tools/hotdoc/default.nix:58]: remove python3Packages

Comment (doronbehar) [reply pkgs/development/tools/hotdoc/default.nix:58]: use `with;`?

Comment (doronbehar) [reply pkgs/development/tools/hotdoc/default.nix:58]: Never mind, it moved to propagated.

File: pkgs/development/tools/hotdoc/default.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/development/tools/hotdoc/default.nix:20]: :+1:.

File: pkgs/development/tools/hotdoc/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/development/tools/hotdoc/default.nix]: Good to know.

File: pkgs/development/python-modules/dbus-deviation/default.nix (hunk 4 above)
Comment (doronbehar) [reply pkgs/development/python-modules/dbus-deviation/default.nix]: :+1:.

File: pkgs/development/tools/hotdoc/default.nix [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  HOTDOC_BUILD_C_EXTENSION = "enabled";
+  nativeBuildInputs = [
+    pkg-config
+    flex
+    cmake
+  ];
+
+  # Cmake is needed to build a c extension CMARK, that's found inside the
+  # source tarball. We add cmake to PATH here, and not in nativeBuildInputs,
+  # since it's not as the main build system of the project.
```
Comment (jtojnar) [inline pkgs/development/tools/hotdoc/default.nix:30]: This sentence is no longer relevant.

File: pkgs/development/tools/hotdoc/default.nix [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  pname = "hotdoc";
+  version = "0.12.2";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "0nypvcqp4b0qfps35f7mz3d90zjh0v5wawpd06ra28q92nxblx17";
+  };
+
+  # https://hotdoc.github.io/installing.html#build-options
+  HOTDOC_BUILD_C_EXTENSION = "enabled";
```
Comment (jtojnar) [inline pkgs/development/tools/hotdoc/default.nix:21]: Could you reorder the expression as in https://discourse.nixos.org/t/document-attribute-ordering-in-package-expressions/4887.

Comment (doronbehar) [reply pkgs/development/tools/hotdoc/default.nix]: Done.

File: pkgs/development/tools/hotdoc/default.nix (hunk 6 above)
Comment (doronbehar) [reply pkgs/development/tools/hotdoc/default.nix]: Rephrased it.

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

File: pkgs/development/python-modules/setuptools-pep8/default.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+  propagatedBuildInputs = [
+    pep8
+  ];
+
+  meta = with stdenv.lib; {
+    description = "This package exposes the pep8 style guide checker as a sub-command of setup.py";
+    homepage = "https://pypi.python.org/pypi/setuptools-pep8";
+    license = licenses.bsd3;
+    maintainers = with lib.maintainers; [ doronbehar ];
```
Comment (jtojnar) [inline pkgs/development/python-modules/setuptools-pep8/default.nix:24]:
```suggestion
    maintainers = with maintainers; [ doronbehar ];
```

Comment (doronbehar) [reply pkgs/development/python-modules/setuptools-pep8/default.nix]: Done.

File: pkgs/development/python-modules/dbus-deviation/default.nix [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "023lkfvbrz3zv68p0qy0xk6r46d7nf4wifwqwmm3gn07d283rfyg";
+  };
+
+  propagatedBuildInputs = [
+    lxml
+  ];
+
+  buildInputs = [
```
Comment (FRidh) [inline pkgs/development/python-modules/dbus-deviation/default.nix:23]: most likely native build inputs

File: pkgs/top-level/python-packages.nix [hunk 10]
```diff
@@ -7470,6 +7474,8 @@ in {
 
   wheel = callPackage ../development/python-modules/wheel { };
 
+  wheezy.template = callPackage ../development/python-modules/wheezy.template { };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix:7477]: normalized name, so `wheezy-template`

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/dbus-deviation/default.nix (hunk 9 above)
Comment (doronbehar) [reply pkgs/development/python-modules/dbus-deviation/default.nix:23]: Right.

File: pkgs/top-level/python-packages.nix (hunk 10 above)
Comment (doronbehar) [reply pkgs/top-level/python-packages.nix]: Fixed.

**Comment by doronbehar** [comment]
I addressed the latest reviews, but maybe let's not merge it yet? Until we are sure it builds gst's docs right. TBH I don't know if I'll find the time for this in the upcoming weeks.

File: pkgs/development/python-modules/setuptools-pep8/default.nix [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "03084wkc6i5i3w2ldpr7mgkdc029p74n0wnskcj2jmq1jiscs87n";
+  };
+
+  propagatedBuildInputs = [
+    pep8
+  ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/setuptools-pep8/default.nix:20]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/python-modules/setuptools-pep8/default.nix [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    description = "This package exposes the pep8 style guide checker as a sub-command of setup.py";
+    homepage = "https://pypi.python.org/pypi/setuptools-pep8";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ doronbehar ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/setuptools-pep8/default.nix:27]:
```suggestion
  };
}
```

File: pkgs/development/tools/hotdoc/default.nix [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+  buildInputs = [
+    glib
+    json-glib
+    libxml2
+  ];
+
+  prePatch =
+    # https://github.com/hotdoc/hotdoc/issues/210
+  ''
```
Comment (SuperSandro2000) [inline pkgs/development/tools/hotdoc/default.nix:53]:
```suggestion
  # https://github.com/hotdoc/hotdoc/issues/210
  prePatch = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
@doronbehar friendly ping

**Comment by doronbehar** [comment]
I don't see a need for this package if we don't use it to build gst's docs, especially not maintainig it. The work here can be used by someone else if they find a need for it.

**Pull request closed without merging**